pkgs/os-specific/linux/sgx/azure-dcap-client/default.nix

```diff
@@ -0,0 +1,89 @@
+{ stdenv
+, fetchFromGitHub
+, fetchurl
+, lib
+, curl
+, nlohmann_json
+, openssl
+, pkg-config
+, linkFarmFromDrvs
+}:
+
+let
+  # Although those headers are also included in the source of `sgx-psw`, the `azure-dcap-client` build needs specific versions
+  headers = linkFarmFromDrvs "azure-dcpa-client-intel-headers" [
+    (fetchurl {
+      url = "https://raw.githubusercontent.com/intel/SGXDataCenterAttestationPrimitives/0436284f12f1bd5da7e7a06f6274d36b4c8d39f9/QuoteGeneration/quote_wrapper/common/inc/sgx_ql_lib_common.h";
+      hash = "sha256-36oxPBt0SmmRqjwtXgP87wOY2tOlbxQEhMZZgjoh4xI=";
+    })
+    (fetchurl {
+      url = "https://raw.githubusercontent.com/intel/linux-sgx/1ccf25b64abd1c2eff05ead9d14b410b3c9ae7be/common/inc/sgx_report.h";
+      hash = "sha256-NCDH3uhSlRRx0DDA/MKhWlUnA1rJ94O4DLuzqmnfr0I=";
+    })
+    (fetchurl {
+      url = "https://raw.githubusercontent.com/intel/linux-sgx/1ccf25b64abd1c2eff05ead9d14b410b3c9ae7be/common/inc/sgx_key.h";
+      hash = "sha256-3ApIE2QevE8MeU0y5UGvwaKD9OOJ3H9c5ibxsBSr49g=";
+    })
+    (fetchurl {
+      url = "https://raw.githubusercontent.com/intel/linux-sgx/1ccf25b64abd1c2eff05ead9d14b410b3c9ae7be/common/inc/sgx_attributes.h";
```

Comment:
since these 3 come from the same repo/HEAD, how about defining and introducing a variable for the commit hash to simplify maintenance?

Reply:
We also thought about that but when we realized that those were touched four years ago for the last time, we decided it's probably not worth it.

Reply:
We could do a sparse checkout to not rely on 3 FODs.

Reply:
See latest push. I’m not sure if that’s really much of an improvement overall.